docs/getting-started/read-access-control.mdx

```diff
@@ -8,19 +8,113 @@ description: Enforce restrictions on reading data from a network.
 
 :::tip
-Setting view procedure privacy is done using the Kwil Gateway.  The gateway is still in private beta.  If you would like access, please contact our team.
+Private RPC mode is not enabled on the Longhorn Testnet. If you would like to test private mode, run a [local node with private RPC enabled](/docs/node/private-rpc#quickstart).
 :::
 
-## Overview
+If your Kwil network is running in private mode, authenticate users by passing a `KwilSigner` to the `call` method.
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+    defaultValue="browser"
+    values={[
+        { label: 'Browser', value: 'browser', },
+        { label: 'NodeJS', value: 'node', },
+        
+    ]}
+>
+<TabItem value='browser'>
+
+```typescript
+import { WebKwil, KwilSigner } from '@kwilteam/kwil-js';
+import { BrowserProvider } from 'ethers';
+
+const kwil = new WebKwil({
+    kwilProvider: "http://localhost:8484",
+    chainId: "your-chain-id",
+});
+
+async function getSigner() {
+    // Prompt user to login with their wallet
+    const provider = new BrowserProvider(window.ethereum);
+    const ethSigner = await provider.getSigner();
+    const ethAddress = await ethSigner.getAddress();
+    return new KwilSigner(ethSigner, ethAddress);
+}
+
+async function authenticatedProcedure() {
+    const kwilSigner = await getSigner();
+
+    const payload = {
+        name: "get_profile",
+        dbid: 'x123abc...'
+    }
+
+    return await kwil.call(payload, kwilSigner); // user will be prompted to authenticate
+}
+```
+
+</TabItem>
+<TabItem value='node'>
+
+```typescript
+import { NodeKwil } from '@kwilteam/kwil-js';
+import { Wallet } from 'ethers';
+
+const kwil = new NodeKwil({
+    kwilProvider: "http://localhost:8484",
+    chainId: "your-chain-id",
+});
+
+async function getSigner() {
+    // Use private key to authenticate
+    const provider = new Wallet("users-private-key");
+    const ethSigner = await provider.getSigner();
+    const ethAddress = await ethSigner.getAddress();
+    return new KwilSigner(ethSigner, ethAddress);
+}
+
+async function authenticatedProcedure() {
+    const kwilSigner = await getSigner();
+
+    const payload = {
+        name: "get_profile",
+        dbid: 'x123abc...'
+    }
+
+    return await kwil.call(payload, kwilSigner);
+}
+```
+</TabItem>
+</Tabs>
+
+## Kwil Gateway
 
 The Kwil Gateway requires a user to sign a message with their [account](/docs/concepts#accounts) keys. Once the signature is verified, the gateway will return the user a cookie to be passed with all subsequent requests.
 
+The main difference between Private RPC and Kwil Gateway is that Private RPC requires the user to sign a message ***for each call request***, whereas the Kwil Gateway enables a user to ***sign in once*** and call multiple procedures.
```

Comment:
Why is this info in the explanation for `Kwil Gateway`? It should be above where we introduce the two types. Also, instead of "sign once and call multiple procedures", it is more descriptive to say "sign once and will remain authenticated for a period of time using a cookie".